src/SDKs/Azure.Base/data-plane/Azure.Base.Tests/ExponentialPolicyTest.cs

```diff
@@ -0,0 +1,105 @@
+using System.Threading.Tasks;
+using Azure.Base.Http.Pipeline;
+using Azure.Base.Testing;
+using Azure.Base.Tests.Testing;
+using NUnit.Framework;
+
+namespace Azure.Base.Tests
+{
+    public class ExponentialPolicyTest: RetryPolicyTestBase
+    {
+        [Test]
+        public async Task WaitsBetweenRetries()
+        {
+            var policy = new ExponentialRetryPolicyMock(retriableCodes: new [] { 500 }, delay: TimeSpan.FromSeconds(1), maxDelay: TimeSpan.FromSeconds(10));
+            var mockTransport = new MockTransport();
+            var task = SendRequest(mockTransport, policy);
+
+            await mockTransport.RequestGate.Cycle(new MockResponse(500));
+
+            var delay = await policy.DelayGate.Cycle();
+            CheckExpectedRange(TimeSpan.FromSeconds(1), delay);
+
+            await mockTransport.RequestGate.Cycle(new MockResponse(200));
+
+            var response = await task.TimeoutAfterDefault();
+            Assert.AreEqual(200, response.Status);
+        }
+
+        [Test]
+        public async Task WaitsSameAmountEveryTime()
+        {
+            var policy = new ExponentialRetryPolicyMock(retriableCodes: new [] { 500 }, maxRetries: 4, delay: TimeSpan.FromSeconds(1), maxDelay: TimeSpan.FromSeconds(10));
+            var mockTransport = new MockTransport();
+            var task = SendRequest(mockTransport, policy);
+            var expectedDelaysInSeconds = new int[] { 1, 2, 4, 8 };
+
+            await mockTransport.RequestGate.Cycle(new MockResponse(500));
+
+            for (int i = 0; i < 4; i++)
+            {
+                var delay = await policy.DelayGate.Cycle();
+                CheckExpectedRange(TimeSpan.FromSeconds(expectedDelaysInSeconds[i]), delay);
+
+                await mockTransport.RequestGate.Cycle(new MockResponse(500));
+            }
+
+            var response = await task.TimeoutAfterDefault();
+            Assert.AreEqual(500, response.Status);
+        }
+
+        [Test]
+        public async Task RespectsMaxDelay()
+        {
+            var policy = new ExponentialRetryPolicyMock(retriableCodes: new [] { 500 }, maxRetries: 6, delay: TimeSpan.FromSeconds(1), maxDelay: TimeSpan.FromSeconds(5));
+            var mockTransport = new MockTransport();
+            var task = SendRequest(mockTransport, policy);
+            var expectedDelaysInSeconds = new int[] { 1, 2, 4, 5, 5, 5 };
+
+            await mockTransport.RequestGate.Cycle(new MockResponse(500));
+
+            for (int i = 0; i < 6; i++)
+            {
+                var delay = await policy.DelayGate.Cycle();
+                CheckExpectedRange(TimeSpan.FromSeconds(expectedDelaysInSeconds[i]), delay);
+
+                await mockTransport.RequestGate.Cycle(new MockResponse(500));
+            }
+
+            var response = await task.TimeoutAfterDefault();
+            Assert.AreEqual(500, response.Status);
+        }
+
+        private void CheckExpectedRange(TimeSpan expected, TimeSpan actual)
```

Comment:
nit: In the `RetryPolicyBase`, you're using a pattern of naming `AssertXXX` which, to me, is a better indicator of context.   It clearly indicates that the method being called contains assertions that may impact the test.  You may want to consider renaming this to `AssertExpectedRange` to keep to the base pattern and offer additional context within the tests using it.